src/databases/Silo/avtSiloFileFormat.C

```diff
@@ -10387,7 +10403,80 @@ avtSiloFileFormat::ReadInConnectivity(vtkUnstructuredGrid *ugrid,
                     vtk_zonetype != -1)
                 {
                     *nl++ = shapesize;
-                    for (k = 0 ; k < (size_t)shapesize ; k++)
+                    int nblinnod = 0 ;
+#ifdef DB_ZONETYPE_QUAD_BEAM
+                    // Handle quadratic elements assuming the first nodes are in Silo convention.
+                    // This is to be more easily compatible with an external face extractor that would consider only
+                    // the first linear nodes. This is not an issue for both TETs and HEXs.
+                    switch (vtk_zonetype)
+                    {
+                    case VTK_QUADRATIC_WEDGE:
+                    {
+                        nblinnod = 6 ;
+                        vtkIdType vtk_wedge[nblinnod];
+                        TranslateSiloWedgeToVTKWedge(nodelist, vtk_wedge);
+                        for (k = 0 ; k < nblinnod ; k++)
+                        {
+                            *nl++ = vtk_wedge[k]-origin;
+                        }
+                        break ;
+                    }
+                    case VTK_QUADRATIC_PYRAMID:
+                    {
+                        nblinnod = 5 ;
+                        vtkIdType vtk_pyramid[nblinnod];
+                        TranslateSiloPyramidToVTKPyramid(nodelist, vtk_pyramid);
+                        for (k = 0 ; k < nblinnod ; k++)
+                        {
+                            *nl++ = vtk_pyramid[k]-origin;
+                        }
+                        break ;
+                    }
+                    case VTK_QUADRATIC_TETRA:
+                    {
+                        nblinnod = 4 ;
+                        // Apply the same logic for Silo quadratic tetras : make sure linear nodes are in the Silo convention,
+                        // i.e. not the VTK one.
+                        // Practically, this means that the user may input tetras into Silo using VTK convention without issue.
```

Comment:
While that is true, it is true only for VisIt. Other downstream Silo consumers may not be so smart :wink: That said, logic and reordering operations like this should really have been put directly into Silo library so that every consumer could easily take advantage of it if they wish. Hindsight is so much clearer than foresight.

Reply:
I'm not sure to understand ; the current logic is VisIt handles this connectivity while it could have been a service of Silo. But I do not want to make too extensive changes to both VisIt and Silo, hence my conservative approach.

Reply:
Oh, yes. Sorry @cessenat. My remark was not very clear, was it! 😞 

You are correct. You do not need to put anything into Silo for this.

My remark was intended more as a note to myself :wink:.